### PR TITLE
Cleanup some Kconfig conditionals and dependencies

### PR DESCRIPTION
**Pull request by nashif**
Make sure the .config does not have unrelated and stray configs that are not being enabled explicitly.

### CODE REVIEW

File: subsys/mgmt/Kconfig [hunk 1]
```diff
@@ -59,6 +59,7 @@ config MCUMGR_SMP_UART_MTU
 
 endif
 
+if MCUMGR
```
Comment (nordic-krch) [inline subsys/mgmt/Kconfig]: aren't all options in this file depending on MCUMGR?

Comment (nashif) [reply subsys/mgmt/Kconfig]: No, not those being ifdefed at least.

File: subsys/dfu/Kconfig [hunk 2]
```diff
@@ -30,6 +30,7 @@ config MCUBOOT_IMG_MANAGER
 	  Enable support for managing DFU image downloaded using mcuboot.
 endchoice
 
+if IMG_MANAGER
```
Comment (MaureenHelm) [inline subsys/dfu/Kconfig]: I think this can move up to line 18

File: subsys/mgmt/Kconfig [hunk 3]
```diff
@@ -82,4 +83,5 @@ config MCUMGR_BUF_USER_DATA_SIZE
 	  Different mcumgr transports impose different requirements for this
 	  setting.  A value of 4 is sufficient for UART, shell, and bluetooth.
 
+endif
```
Comment (MaureenHelm) [inline subsys/mgmt/Kconfig]: nit: `endif # MCUMGR`

**Comment by nashif** [comment]
@MaureenHelm updated

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**